pandas/core/generic.py

```diff
@@ -11222,6 +11222,8 @@ def logical_func(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs
                 raise NotImplementedError(
                     "Option bool_only is not implemented with option level."
                 )
+            if isinstance(self, pd.Series) and self.dtype.name == "boolean":
```

Comment:
pd.Series -> ABCSeries

is_bool_dtype(self.dtype)

Reply:
@jbrockmendel 
We need to distinguish between bool and boolean and is_bool_dtype() returns true in both cases.

Reply:
makes sense, thanks

Reply:
>  We need to distinguish between bool and boolean and is_bool_dtype() returns true in both cases.

Might be worth something like `is_bool_na_dtype`